### PR TITLE
Add enqueue and dequeue commands

### PR DESCRIPTION
**Pull request by wongsm7**
Improvement Intended
Implements #34 and #56

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -29,28 +32,75 @@
...
+
+    //=========== QueueManager ==================================================================================
+
+    @Override
+    public QueueManager getQueueManager() {
+        return queueManager;
+    }
+
+    @Override
+    public void removePatient(ReferenceId target) {
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/model/ModelManager.java]: Please give more descriptive names to your functions. Possibility, removeFromQueue(...)

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -29,28 +32,75 @@
...
+        queueManager.removePatient(target);
+    }
+
+    @Override
+    public void removePatient(int index) {
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/model/ModelManager.java]: Same here, please consider "removeFromQueue(...)"

File: src/main/java/seedu/address/model/ModelManager.java [hunk 3]
```diff
@@ -29,28 +32,75 @@
...
+        queueManager.removePatient(index);
+    }
+
+    @Override
+    public void addPatient(ReferenceId id) {
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/model/ModelManager.java]: "A tactical dot."

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -29,28 +32,75 @@
...
+        queueManager.addPatient(id);
+        updateFilteredReferenceIdList(PREDICATE_SHOW_ALL_ID);
     }
 
+    @Override
+    public void next(int index) {
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/model/ModelManager.java]: "A tactical dot."

File: src/main/java/seedu/address/model/ModelManager.java [hunk 5]
```diff
@@ -29,28 +32,75 @@
...
+        queueManager.serveNext(index);
+    }
+
+    @Override
+    public boolean hasId(ReferenceId id) {
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/model/ModelManager.java]: "A tactical dot."

File: src/main/java/seedu/address/model/ModelManager.java [hunk 6]
```diff
@@ -29,28 +32,75 @@
...
+        requireNonNull(id);
+        return queueManager.hasId(id);
+    }
+
+    @Override
+    public void addRoom(ReferenceId id) {
+        queueManager.addRoom(id);
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/model/ModelManager.java]: Could you throw an UnsupportedOperationException for now? As you have yet to implement the underlying logic.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 7]
```diff
@@ -29,28 +32,75 @@
...
+    }
+
+    @Override
+    public void removeRoom(int index) {
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/model/ModelManager.java]: "A tactical dot."

File: src/main/java/seedu/address/model/QueueList.java [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    public boolean hasId(ReferenceId id) {
+        requireNonNull(id);
+        return ids.contains(id);
+    }
+
+    /**
+     * Adds a person to the queue
+     * The person must not already exist in the queue.
+     */
+    public void addPatient(ReferenceId id) {
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/model/QueueList.java]: "A tactical dot."

File: src/main/java/seedu/address/model/QueueList.java [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+        ids.add(id);
+    }
+
+    public void setIds(List<ReferenceId> ids) {
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/model/QueueList.java]: "A tactical dot."

File: src/main/java/seedu/address/model/QueueList.java [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+        this.ids.setIds(ids);
+    }
+
+    /**
+     * Removes {@code key} from this {@code AddressBook}.
+     * {@code key} must exist in the address book.
+     */
+    public void removePatient(ReferenceId id) {
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/model/QueueList.java]: "A tactical dot."

File: src/main/java/seedu/address/model/QueueList.java [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+        ids.remove(id);
+    }
+
+    public void removePatient(int index) {
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/model/QueueList.java]: "A tactical dot."

File: src/main/java/seedu/address/model/QueueList.java [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+        ids.remove(index);
+    }
+
+    public ReferenceId getFirst() {
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/model/QueueList.java]: Mind renaming this to "peek()"

**Comment by CarbonGrid** [comment]
It shouldn't overwrite the master's AutoCompleter I think.

**Pull request merged**